Odroid h4 ultra tests

miczyg1 opened the pull request:
Introduce to new tests: IBECC and ODROID netcard.

The netcard test is pretty straightforward. It checks the performance of two pairs of the network ports as in regular ETHPERF test.

IBECC simply verifies whether the error in dmesg appears when IBECC is disabled/enabled or not.

I also prepared the ODROID H4 Ultra target (the IPs are temporary, they will get updated when the platform gets installed in the lab).

miczyg1 committed:
dasharo-compatibility/network-speed.robot: Move keywords to lib

The keywords are generic for network performance testing. They can
be reused by other tests, e.g. incoming ODROID netcard test.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/odroid-netcard.robot: Add netcard test

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/memory-ibecc.robot: Add IBECC tests

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs: Add ODROID H4 Ultra

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-performance/fast-boot.robot: Make the test work with telnet

The test assumed that the device is already booted to an OS, which probably
is true for laptops. Add necessary Power On's and initial booting to OS
on the beginning of the test to handle a case where the serial connection
can be used to navigate and select OSes to boot.

Also do not log in to OS again after "Measure FW Boot Time On Linux",
because this keyword keeps the connection to the OS.

TEST=Run the test and PASS it on ODROID H4 Ultra

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented:
test results:
https://github.com/Dasharo/coreboot/pull/682#issuecomment-2962095821
https://github.com/Dasharo/coreboot/pull/680#issuecomment-2962543047

miczyg1 commented:
I am struggling a bit with suspend tests. Manually it works, but in tests the output is read in wrong places and the test assume that SSH is used to connect, because after waking up `Login To Linux` is used. But with telnet we still are in the logged system... This breaks many tests it seems, because fastboot tests assumed similar things...

```diff
diff --git a/lib/sleep-lib.robot b/lib/sleep-lib.robot
index 71a200c2..c093116e 100644
--- a/lib/sleep-lib.robot
+++ b/lib/sleep-lib.robot
@@ -93,10 +93,12 @@ Perform Suspend Test Using FWTS
     ...    test by using Firmware Test Suite tool
     [Arguments]    ${test_duration}=40
     ${is_suspend_performed_correctly}=    Set Variable    ${FALSE}
+    Read From Terminal
     Write Into Terminal    fwts s3 -f -r /tmp/suspend_test_log.log
     Sleep    ${test_duration}s
-    Login To Linux
-    Switch To Root User
+    Read From Terminal
+    Write Bare Into Terminal    ${ENTER}
+    Read From Terminal Until Prompt
     ${test_result}=    Execute Linux Command    cat /tmp/suspend_test_log.log
     TRY
         Should Contain    ${test_result}    0 failed
```

miczyg1 committed:
lib/sleep-lib.robot: Make sleep tests work with telnet

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
test_cases.json: Add new test cases

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

macpijan commented:
> I am struggling a bit with suspend tests.

Still struggling here?

Generally, changes look fine. We should:
- [ ] Get rid of WIP commits prior merging
- [ ] make a decision on sleep tests - I am sure some of us have been running these as well, we should discuss how to apporach it

miczyg1 committed:
platform-configs/odroid-h4: Do not verify FW version after flashing

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/custom-network-boot-entries.robot: Ensure netboot is enabled

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Add CPU temperature threshold

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Update release date

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
variables.robot: Fix the configs for ODROID platforms

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/miniPCIe-slot-verification.robot: Fix skip

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/nvme-support.robot: Use configs to get disk name

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs: Drop unused CLEVO_DISK

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Synchronize configs

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Enable Windows tests

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
os-config/ansible/os-specific-tasks/201.yaml: Add docker installation

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
scripts/lib/robot.sh: Fix typo in directory name

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Disable fan speed measurements

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/usb-hid-msc-lib.robot: Fix condition for mounting USB

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/device-detection.robot: Close connections at the end of teardown

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/device-detection.robot: Add proper checks for suite skip

Judging by the content of test cases, the suite depends on the
SSH connection method and the DCU backed for option manipulation.
It is not suitable for runnign in other configuration, like
telnet connection method and option management via firmware setup.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Update release version

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/efibootmgr.robot: Make test suitable for telnet platforms

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
keywords.robot: Wait for the power to stabilize before pressing power button

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
util/basic-platform-setup.robot: Mark suite as automated

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
test_cases.json: Sort new tests by ID

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/audio-subsystem.robot: Use generic keyword for command

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/cpu-fan-speed-measure.robot: Skip suite earlier to save time

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/network.robot: Make intervals larger for Get Hostname Ip

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Disable hybrid core support

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/network.robot: Add variable for Get Ip interval

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
scripts/regression-rerun-failed.sh: Add script to rerun failed tests

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/efibootmgr.robot: Ensure network boot is disabled initially

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Fix CPU temperature threshold

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/power-after-fail-lib.robot: Fix import failure

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/reset-to-defaults.robot: Skip fan profile test if not supported

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/platform-suspend-and-resume.robot: Check for skip in suite setup

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
compatibility/reset-to-defaults.robot: Skip sleep type test if not supported

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
Deploy UEFI Shell where necessary

Basic platform setup deploye UEFI SHell, however running a whole
suite like comaptibility, e.g. during regression, causes the FW
to eb flashed again and the UEFI Shell is lost right after basic
platform setup finishes its job.

Ensure the UEFI Shell is deployed wherever it is used.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/esp-scanning-lib.robot: Sync filesystem after cleaning ESP

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

philipanda commented on dasharo-compatibility/efibootmgr.robot:
```diff
@@ -31,10 +31,18 @@ EBM001.201 Network Boot enable
 
     Set UEFI Option    NetworkBoot    ${TRUE}
+
+    IF    '${ENV_ID_FEDORA}' in ${TESTED_LINUX_DISTROS}
```
In `EBM001.201` the `.201` means that this test is performed on Ubuntu (Check `docs/adding-and-naming-test-cases.md`). If the test should be performed on Fedora or any other OS, then a new test case must be created

miczyg1 replied:
Yeah, got a bit ahead of myself here. Changed: b8e80db2

philipanda commented on dasharo-compatibility/memory-ibecc.robot:
```diff
@@ -0,0 +1,72 @@
+Set IBECC State
+    [Arguments]    ${target_state}
+    Power On
+    Enter Setup Menu Tianocore
```
I think you could use `Enter Setup Menu Tianocore And Return Construction` instead of a couple of the following lines.

miczyg1 replied:
61e150cc

philipanda commented on dasharo-compatibility/memory-ibecc.robot:
```diff
@@ -0,0 +1,72 @@
+    ${setup_menu}=    Parse Menu Snapshot Into Construction    ${out}    3    1
+    ${dasharo_menu}=    Enter Dasharo System Features    ${setup_menu}
+    ${memory_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Memory Configuration
+    ${status}=    Set Option State    ${memory_menu}    Memory In-Band ECC    ${target_state}
```
If the intention is only to set a new Setup menu option, then I would suggest describing the path to it in `lib/bios/menus.py` and using the `Set UEFI Option` keyword instead of re-implementing this logic again, unless I can't see some non-standard step that is required here

miczyg1 replied:
Yes, I probably wrote this when these keywords were not available.

miczyg1 replied:
61e150cc

philipanda commented on dasharo-compatibility/nvme-support.robot:
```diff
@@ -34,7 +34,16 @@ NVM001.001 NVMe support in firmware
     Power On
     ${out}=    Enter Boot Menu Tianocore And Return Construction
-    Should Contain    ${out}    ${CLEVO_DISK}
+    ${ssd_list}=    Get Current CONFIG List Element    Storage_SSD
```
`variables.robot` is being deprecated https://github.com/Dasharo/open-source-firmware-validation/issues/674
I would suggest not adding new dependencies on it if possible.
I appreciate the more verbose messaging during this test though

miczyg1 replied:
I see. Any suggestion how to get rid of `CLEVO_DISK` used on every platform just to validate NVMe visibility in firmware? Hwo do you plan to support multiple boards that will source the same configs (e.g. two ODROIDs H4+ ) but with different disks connected?

Is it even good to have a variable with clevo in name? Who made this up?

philipanda replied:
It probably was added by someone who at the time did not fully understand what the variables.robot file was used for? I think ideally the information about hardware configuration of a given DUT should be outsourced to snipeit and allow to override it using environment variables.

There are multiple platform-configs variables that really shouldn't be there, because as you said, another DUT might very well have a different disk model attached during testing. Unless we are enforcing the tests to be run using a well defined configuration the platform-configs need to be modified on every release because of that.

Helping with this is not the reason for this PR, so from my side using the variables.robot here is fine. Unless you have a vision of how to solve this better without diving deeper into the variables.robot/platform-configs situation, then that would be great

miczyg1 replied:
>  Unless we are enforcing the tests to be run using a well defined configuration the platform-configs need to be modified on every release because of that.

I still believe we are running tests na a well defined configuration, no? Otherwise, why would we describe HW matrix on dasharo docs?

philipanda replied:
> I still believe we are running tests na a well defined configuration, no? 

That's what I would like to believe, but I don't think everyone makes sure to follow it 100%.
Maybe if the DUT's weren't shuffled regularily and the precise set of models of resources like keyboards and usb-drives was easier to find and complete, then it wouldn't be an issue. In practice, I doubt any tester would invalidate a test result just because the model name of used usb-keyboard was different than expected.
Maybe they should?
Anyhow that's not related to the changes, so from me the conversation is resolved. We might continue the topic of the hardware matrix and variables.robot somewhere else

philipanda commented on keywords.robot:
```diff
@@ -1522,3 +1540,13 @@ Should Contain All
         Should Contain    ${string}    ${substring}
     END
+
+Deploy Uefi Shell
```
UEFI Shell should is installed on the system drive on every DUT by the test suite `util/basic-platform-setup.robot`. It should always be run prior to any testing to ensure that all test dependencies are installed on the device. Check https://github.com/dasharo/open-source-firmware-validation?tab=readme-ov-file#basic-platform-setup

I suggest running the suite on your DUT to ensure everything is prepared for testing instead of implementing it in test cases

miczyg1 replied:
That doesn't work.

I know that BPS deploys the UEFI Shell, but as soon as BPS finishes and next suite runs, e.g. dasharo-compatibility, the suite's  `__init__.robot` flashes the firmware again, and the Shell option is already gone.

tl;dr

BPS deploying the  UEFI Shell is useless during regression...

miczyg1 replied:
Also `__init__.robot` isn't the only place where the firmware is flashed, so it is very easy for the UEFi Shell to disappear from boot option list...

philipanda replied:
Hmm, it makes sense. Afterall the bootentry is manually created using efibootmgr, so it won't persist after flashing and the firmware is not discovering this bootable file on its own.

I am not sure what would be needed for the automatic discovery to work. Maybe a new tiny partition for the UEFI shell should be created on the system drive? Ideally all the dependencies should be installed as a preparation step to free the tests from this chore. Maybe we should create an issue to make the installation of UEFI shell during BPS persistent? Until that's solved I think your solution is good.

miczyg1 replied:
https://github.com/Dasharo/open-source-firmware-validation/issues/1059

philipanda commented on lib/sleep-lib.robot:
```diff
@@ -102,7 +102,11 @@ Perform Suspend Test Using FWTS
     END
-    ${test_result}=    Execute Command In Terminal    cat /tmp/suspend_test_log.log
+    # Clean up console before reading file
+    Read From Terminal
```
`Execute Command In Terminal` already flushes the terminal before executing the requested command via serial console using `Telnet.Read`

miczyg1 replied:
It didn't before and I had output from previous command(s) when testing it...

I would have to check again, but right now the suspend tests won't work on Ubuntu, because of upstream ubuntu bug causing the fwts to fail to install: https://bugs.launchpad.net/ubuntu/+source/fwts/+bug/2111123

miczyg1 replied:
I had to change this anyways https://github.com/Dasharo/open-source-firmware-validation/pull/881/commits/ef4592f2d9ef483fe90f344f300c2b19af8411c7

The problem in fact was that the `Execute Command In Terminal` expects the promtp to appear after command is executed. However, the FWTS S3 test will not return the prompt after platform resumes. One has to additionally press ENTER for the prompt to appear again.

philipanda replied:
I wonder if it's specific to the version of FWTS you've had installed here? I had no issue with running suspend tests for NUCBOX v0.9.0 via serial console. The changes look like they won't break anything should the prompt appear normally, so I think it's fair to use them

philipanda commented on os-config/ansible/os-specific-tasks/201.yaml:
```diff
@@ -13,3 +13,34 @@
 - name: Install UEFI Shell
   ansible.builtin.command: /tmp/deploy-shell-efi.sh /tmp/Shell.efi
+
+- name: Install docker required packages
```
Could you also add this to the tasks for Fedora? (202.yaml)

miczyg1 replied:
I have no way to test it on ODROID, sorry. So to avoid breaking something, I won't add it.

philipanda replied:
So maybe let's create an issue for that? Docker is definitely a dependency of some tests and it's clearly an oversight that it's not installed during BPS

miczyg1 replied:
https://github.com/Dasharo/open-source-firmware-validation/issues/1060

philipanda commented on platform-configs/odroid-h4-plus.robot:
```diff
@@ -42,6 +42,7 @@ ${AUDIO_SUBSYSTEM_SUPPORT}=                     ${TRUE}
 ${EXTERNAL_HEADSET_SUPPORT}=                    ${TRUE}
 ${EXTERNAL_HDMI_DISPLAY_SUPPORT}=               ${TRUE}
 ${EXTERNAL_DISPLAY_PORT_SUPPORT}=               ${TRUE}
+${FAN_SPEED_MEASURE_SUPPORT}=                   ${FALSE}
```
It's already set to ${FALSE} in `default.robot`

miczyg1 replied:
6012420f

philipanda commented on platform-configs/odroid-h4-ultra.robot:
```diff
@@ -0,0 +1,149 @@
+${EXTERNAL_HEADSET_SUPPORT}=                    ${TRUE}
+${EXTERNAL_HDMI_DISPLAY_SUPPORT}=               ${TRUE}
+${EXTERNAL_DISPLAY_PORT_SUPPORT}=               ${TRUE}
+${FAN_SPEED_MEASURE_SUPPORT}=                   ${FALSE}
```
Already in `default.robot`

miczyg1 replied:
6012420f

miczyg1 committed:
dasharo-compatibility/DTS: Skip flashing if FW variable not set

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/dasharo-tools-suite.robot: Fix internal flashing

"Flash Via Internal Programmer" sends a file to DUT from host. However,
the test downloads th binary with wget prior to the keyword execution.
Skip the wrapper and call "Flash Via Internal Programmer With Args"
directly.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-security/tpm: Fix TPM tests

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-security/verified-boot.robot: Sync filesystem before power cycling

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/audio-subsystem.robot: Fix headphones Windows tests

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/audio-subsystem.robot: Make audo devices platform-specific

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/bios/menus.robot: Fix variable type

Fixes the error of variable value not a list or list-like.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-security/verified-boot.robot: Drop VBO011.001 duplication

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/sleep-lib.robot: Fix suspend test on Telnet

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Remove obsolete override

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

philipanda commented on dasharo-compatibility/device-detection.robot:
```diff
@@ -19,11 +19,15 @@ Resource            ../keys.robot
+...                     Skip If    '${OPTIONS_LIB}' != 'options-lib_dcu'
+...                     AND
+...                     Skip If    '${DUT_CONNECTION_METHOD}' == 'telnet'
+...                     AND
```
Why isn't it possible to perform the tests using the Setup Menu?
The option to disable the USB stack should be perfectly possible to be changed using `options-lib_uefi-setup-menu`
```
    "UsbDriverStack": [
        "Dasharo System Features",
        "USB Configuration",
        "Enable USB stack",
    ],
    ```

philipanda replied:
I see, it would be possible had the test been written correctly:
```
Login To Linux With Root Privileges
Set UEFI Option UsbDriverStack Disabled
Execute Reboot Command
Sleep 10s
```
https://github.com/Dasharo/open-source-firmware-validation/issues/1061

philipanda commented on dasharo-compatibility/odroid-netcard.robot:
```diff
@@ -22,12 +23,21 @@ Suite Setup         Run Keyword
 *** Test Cases ***
-STUB001.001 External flash explicit
-    Flash Device Via External Programmer
+NETCARD001.201 Check Performance of 2.5G Wired Network Interface On Netcard (Ubuntu)
```
That's some wild diff from GitHub, I must say :laughing: 
The STUB test suite appears at the osfv-dashboard, so we can't delete it just like that, even if it is arguably useful, and even if so, then shouldn't be a part of dasharo-compatibility.

Why not adjust the ETHPERF test suite and create a new one? They seem to do  the same thing apart from some manual preparation in ETHPERF001.201, which could be be skipped conditionally  on odroid if that's not needed

miczyg1 replied:
> Why not adjust the ETHPERF test suite and create a new one? They seem to do the same thing apart from some manual preparation in ETHPERF001.201, which could be be skipped conditionally on odroid if that's not needed

The way it is tested resembles ETHPERF, but it just uses a performance test to prove compatibility (i.e. that the netcard works correctly). That is why I decided for it to be separate.

miczyg1 replied:
Dropped the commit that removed flash stub.

philipanda replied:
> That is why I decided for it to be separate.

> The way it is tested resembles ETHPERF, but it just uses a performance test to prove compatibility 

Yet it's name's "Check Performance of..."?
While I perfectly understand making one test for one thing, I hate how much time that often wastes in regressions. The main difference in the suites is the name. Why couldn't we be assuming, that if the netcard performance is fine, then the netcard must surely work properly?

miczyg1 replied:
> While I perfectly understand making one test for one thing, I hate how much time that often wastes in regressions. The main difference in the suites is the name. Why couldn't we be assuming, that if the netcard performance is fine, then the netcard must surely work properly?

I guess we can. Currently the only problem is that ETHPERF tests can verify only 1 ETH pair. But I need to test 2 pairs. Any suggestions how to do this so we could resolve this thread? @philipanda

miczyg1 replied:
Also it seems the check for empty list of ETH ports fails:

<img width="1912" height="88" alt="image" src="https://github.com/user-attachments/assets/4ab483f5-6831-457e-8c68-8db612cc2f3b" />

philipanda replied:
>  Currently the only problem is that ETHPERF tests can verify only 1 ETH pair. But I need to test 2 pairs. Any suggestions how to do this so we could resolve this thread?

I think it would be enough to add more pairs to this `${ETH_PERF_PAIR_10_G}`  variable and iterate over pairs in the list
```robot
    ${eth_1}=    Get From List    ${ETH_PERF_PAIR_10_G}    0
    ${eth_2}=    Get From List    ${ETH_PERF_PAIR_10_G}    1
 ```
 something like:
 ```
 for i=0; i<len(ETH_PERF_PAIR_10_G); i+=2 {
     perform_test(ETH_PERF_PAIR_10_G[i], ETH_PERF_PAIR_10_G[i+1]
 }
 ```
 
 The `${target_bitrate}` which is currently hardcoded in the test could also get parametrized. One ETHPERF test uses `2.35`, just like your `NETCARD`, but another one uses `9.35` for no reason apparent to me

philipanda replied:
> Also it seems the check for empty list of ETH ports fails:

Seems like an issue caused by the quotes surrounding the variables. Their values are expanded into literals before being passed as a string parameter to `Depends On`. This string representation, which contains the additional quotes is then `Evaluate`'d, which fails, because the pairing of the quotes is now broken as i your screenshot. Any test using `Depends On` previously has been passing plain variables, not logic expressions, so it worked. I think removing the quotes should fix this bug
```robot
    Depends On    ${ETH_PORTS} != @{EMPTY}
```

miczyg1 replied:
Yeah I have fixed ETH_PORTS array the way you explained. I wonder how it even worked before.

miczyg1 replied:
@philipanda moved the net card tests to ETHPERF: 16483ed4
Created a new case because the first one is semiauto, while ODROID with netcard will always be able to do fully automated ETHPERF test.

But if you ask me what should be where in this testing framework, I would definitely not add "performance" test to compatibility suite (we have performance suite). And certainly not add `platform-stability.robot` to performance suite (we have stability suite). Someone's had a really good decision process here.

philipanda replied:
Does ODROID ULTRA have other network interfaces that we would like to test other than the Net Card 2 ones?
If `ETHPERF003` and `ETHPERF001` tests are mutually exclusive, I'd definitely merge them. Otherwise, why not create separate tests for ODROID ULTRA for every ethernet-related suite?

I think the `NETCARD` tests were not pushed to osfv-dashboard, so a simple replacement in `test_cases.json` should also be fine instead of using `changed_to` fields.

miczyg1 replied:
Merged into ETHPERF001: 8ac0fa8a

philipanda replied:
Great, that's much better :+1:

philipanda commented on dasharo-performance/fast-boot.robot:
```diff
@@ -20,15 +20,16 @@ FBT001.201 Fast Boot Reduces Boot Time
     Log To Console    \nMeasuring boot time without Fast Boot\n
-    Login To Linux
-    Switch To Root User
+    # We should still be in the OS as root after Measure FW Boot Time On Linux
```
That should be a commit message, not a comment right?
Without context I wouldn't understand why this comment is here

miczyg1 replied:
Extended the commit message: 8613c268

philipanda commented on lib/sleep-lib.robot:
```diff
@@ -89,15 +89,19 @@ Perform Suspend Test Using FWTS
         Switch To Root User
     END
-    ${test_result}=    Execute Command In Terminal    cat /tmp/suspend_test_log.log
+    ${test_result}=    Execute Linux Command    cat /tmp/suspend_test_log.log
```
Why was that required?
We would like to transfer from using `keywords.robot` keywords to the ones defined in `lib` and ideally remove `keywords.robot` altogether

miczyg1 replied:
TBH, I don't know why this keyword changed. Reverted: 441aa43c4e04f11ba2c61f75aa949972f750a7e0

philipanda replied:
Maybe you're using some linter and the `terminal` lib was not imported, so it changed it to a similar keyword from `keywords.robot`?

philipanda commented on platform-configs/odroid-h4-plus.robot:
```diff
@@ -10,24 +10,24 @@ ${FLASH_LENGTH}=                                ${TBD}
 ${CPU}=                                         Intel(R) N97
 ${POWER_CTRL}=                                  RteCtrl
-${FLASH_VERIFY_METHOD}=                         tianocore-shell
+${FLASH_VERIFY_METHOD}=                         none
```
Why not leave `tianocore-shell` if you've made a workaround that restores UEFI Shell?
That way we are not verifying if the correct binary is really flashed at  `Prepare Platform`

miczyg1 replied:
The deploy shell keyword assumes there is an OS installed on the platform. However, the tianocore-shell flash verify will instantly try to boot the shell right after flashing. So that method is not suitable IMHO. It was good as long as Shell was included by default.

philipanda commented on scripts/regression-rerun-failed.sh:
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2024 3mdeb <contact@3mdeb.com>
```
It's 2025

miczyg1 replied:
6c8084c1

miczyg1 committed:
scripts/regression.sh: Run stability suite too

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4-ultra.robot: Sync DTS E2E variables

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4: Sync DMI info

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-stability/tpm-detect.robot: Drop check for NVME in TPM detection

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility: Fix list comparison in Ethernet tests

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/power-after-fail.robot: Fix powering on based on var state

When power after fail test are executed, the state of power after fail variable
is changed. The Power On keyword does not take it int oaccount and uses only
the default state of the variable. This results in obsolete power button presses
which often lead to powering off the platform right after PSU is powered on.
Because of that the tests fail due to power LED not being lit up.

Define another variable that is updated runtime and represents current state of
the power after faile variable.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
keywords: Fix usage in teardown

Fixes the problem: 'Set Tags' cannot be used in suite teardown.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs/odroid-h4-ultra.robot: Update Ethernet info

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
util/basic-platform-setup.robot: Add minimal-regression tag

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/secure-boot-lib.robot: Fix launching UEFI Shell

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
lib/linux.robot: Add MKTME to dmesg error allowlist

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
variables.robot: Add ODROID H4 Ultra

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility: Move netcard test to ETHPERF

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

philipanda commented on dasharo-compatibility/network-speed.robot:
```diff
@@ -76,12 +77,19 @@ ETHPERF001.201 Check Performance of 2.5G Wired Network Interface (Ubuntu)
+    # ODROID
+    IF    ${ODROID_NETCARD_SUPPORT}
+        # Test second pair of network ports on netcard
+        Configure Network Interfaces For Testing    enp6s0    enp7s0
```
Why hardcode the interface names?

miczyg1 replied:
Removed: 8ac0fa8a

philipanda commented on dasharo-compatibility/device-detection.robot:
```diff
@@ -21,13 +21,19 @@ Suite Setup         Run Keywords
+...                     AND
 ...                     Set UEFI Option    UsbDriverStack    Enabled
+...                     AND
+...                     Log Out And Close Connection
```
Won't double `Log Out And Close Connection` break the teardown? It's already there a couple lines above

miczyg1 replied:
Good catch. Probably bad rebase from me. I have to fix it. But notice how the close connection happens before `Set UEFI Option`. The keyword had no chance to pass after closing connections..

miczyg1 replied:
dee1fbc0

miczyg1 committed:
dasharo-compatibility/network-speed.robot: merge netcard test with ETHPERF001

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
dasharo-compatibility/device-detection.robot: Fix duplicated close connection

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
platform-configs: Drop unused ODROID_NETCARD_SUPPORT

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
